src/main/java/duke/Command.java

```diff
@@ -0,0 +1,11 @@
+package duke;
+
+/**
+ * Abstract class representing the commands supported by Duke.
+ * All commands will interact with the current task list, UI and
+ * tasks saved to disk in some way, these behaviours are to be
+ * implemented by subclasses of this class.
+ */
+public abstract class Command {
```

Comment:
I feel that all the `Command`-related classes can be put in one package, called commands, what do you think? Same for other Java files, such as `Task` and exceptions.